contents/projectManagement/testDrivenDevelopment.md

```diff
@@ -0,0 +1,169 @@
+
+## Limitations
+
+Even with some of the attractive benefits of adopting TDD, there is no one-size-fits-all solution for project design management. Some of the limitations of TDD are: 
+
+### Limitation 1: Strict Emphasis on Communication of Requirements and Specifications
+
+Product/Feature requirements are crucial to teams that employ a TDD-style approach to development. As mentioned earlier, the very first step of the development cycle is to write acceptance tests. If the requirements are not communicated properly, teams will end up constantly having to change their test suites. In extreme scenarios, a re-write may be necessary.
+
+As a result, a lot of man hours can go into making sure the requirements are in order -  consequently, this could result in a slow start to development.
+
+### Limitation 2: Tough to Incorporate into Legacy Projects
+
+<trigger trigger="click" for="modal:index-legacy">Legacy projects</trigger> [^7] are hard to maintain even without having to incorporate TDD into them. As these projects have a history of design and implementation choices made based on the requirements and/or the resources at the time, investing man hours in revamping or "modernizing" them is not always deemed as top priority. 
+
+<modal title="Legacy Code" id="modal:index-legacy">
+Legacy code refers to an application system source code type that is no longer supported. Legacy code can also refer to unsupported operating systems, hardware and formats. In most cases, legacy code is converted to a modern software language and platform. However, to retain familiar user functionality, legacy code is sometimes carried into new environments. 
+</modal>
+
+When you bring TDD into the picture, things get more complex as failing tests are written at first with the assumption that production code will be written to ensure they pass. 
+
+Writing production code in legacy systems with the goal of passing a specific test is harder than it seems. For instance, there may be dependencies between components that need to be "broken" for them to be testable, which could result in  several workarounds. This could potentially add on to the complexity of the project, making it counter-productive as the entire point of the refactoring step is to make the code more readable [^5].
+
+### Limitation 3: High Maintenance
+    
+Maintaining test-suites becomes a part of the overall overhead of the project. Since TDD-style approaches rely heavily on tests, there is a need for the tests to be in proper shape in order to prevent a false sense of correctness. 
+
+TDD also has a steep learning curve. The entire process takes time to get accustomed to. Refactoring of code by using good design patterns comes with time, experience, and a lot of practice.
+
+### Limitation 4: Great Importance Given to Unit Tests
```

Comment:
Due to the word choice here, it sounds like a positive thing. `Possibility of over-emphasizing unit tests at the expense of other tests`?